cirq-core/cirq/circuits/circuit_operation_test.py

```diff
@@ -692,6 +692,10 @@ def test_decompose_nested():
         cirq.measure(d, key='md'),
     )
     assert cirq.Circuit(cirq.decompose(final_op)) == expected_circuit
+    # Verify that mapped_circuit gives the same operations.
+    assert list(final_op.mapped_circuit(deep=True).all_operations()) == list(
+        expected_circuit.all_operations()
+    )
```

Comment:
Why can't we check circuit equality here?
```python
assert final_op.mapped_circuit(deep=True) == expected_circuit
```

Reply:
I thought we couldn't (had some weird test failures earlier), but we totally can. Done.